python/grass/script/db.py

```diff
@@ -233,23 +246,139 @@ def db_table_in_vector(table, mapset=".", env=None):
+
+        :param str driver_name: DB driver name
+        :param str database: database name
+        """
+        self._driver_name = driver_name
+        self._database = database
+        self._import_c_funcs()
+
+    def _import_c_funcs(self):
+        """Import C functions"""
+        try:
+            from grass.lib.dbmi import (
+                db_begin_transaction,
+                db_close_database_shutdown_driver,
+                db_commit_transaction,
+                db_execute_immediate,
+                db_free_string,
+                db_init_string,
+                db_set_string,
+                db_start_driver_open_database,
+                dbString,
+                DB_OK,
+            )
+            from grass.lib.gis import G_gisinit
+            from grass.lib.vector import (
+                Map_info,
+                Vect_subst_var,
+            )
 
-def db_commit_transaction(driver):
-    """Commit transaction.
+            self._c_funcs = {
+                "db_begin_transaction": db_begin_transaction,
+                "db_execute_immediate": db_execute_immediate,
+                "db_free_string": db_free_string,
+                "db_init_string": db_init_string,
+                "db_close_database_shutdown_driver": db_close_database_shutdown_driver,
+                "db_commit_transaction": db_commit_transaction,
+                "db_set_string": db_set_string,
+                "db_start_driver_open_database": db_start_driver_open_database,
+                "dbString": dbString,
+                "DB_OK": DB_OK,
+                "G_gisinit": G_gisinit,
+                "Map_info": Map_info,
+                "Vect_subst_var": Vect_subst_var,
+            }
+        except (ImportError, OSError, TypeError) as e:
+            fatal(_("Unable to import C functions: {e}").format(e))
+
+    def _init_driver(self):
+        """Init DB driver"""
+        map = self._c_funcs["Map_info"]()
+        self._pdriver = self._c_funcs["db_start_driver_open_database"](
+            self._driver_name,
+            self._c_funcs["Vect_subst_var"](self._database, byref(map)),
+        )
+        if not self._pdriver:
+            fatal(
+                _("Unable to open database <{db}> by driver <{driver}>.").format(
+                    db=self._database, driver=self._driver_name
+                )
+            )
 
-    :return: SQL command as string
-    """
-    if driver in ("sqlite", "pg", "mysql"):
-        return "COMMIT"
-    return ""
+    def _begin_transaction(self):
+        """Begin DB transaction."""
+        self._c_funcs["G_gisinit"]("")
+
+        self._init_driver()
+
+        ret = self._c_funcs["db_begin_transaction"](self._pdriver)
+        if ret != self._c_funcs["DB_OK"]:
+            self._c_funcs["db_close_database_shutdown_driver"](self._pdriver)
+            fatal(
+                _(
+                    "Error while start database <{db}> transaction by"
+                    " driver <{driver}>."
```

Comment:
The `start` sounds weird. Is `starting` the correct wording?
```suggestion
                    "Error while starting database <{db}> transaction by"
                    " driver <{driver}>."
```